controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -604,6 +624,177 @@ func getFailureDomainKeyFromStorageClassParameter(sc *storagev1.StorageClass) st
 	}
 }
 
+func (r *OCSInitializationReconciler) reconcileOCSServerService(initialData *ocsv1.OCSInitialization) (reconcile.Result, error) {
+	/*
+		OCS API Server ServiceType follows the following level of priority:
+			1. Value set in OCSInitialization
+			2. Value set in InternalStorageCluster
+		This helps to maintain backward compatibility
+	*/
```

Comment:
// The ServiceType for the OCS API Server is determined based on the following priority:
// 1. The value set in OCSInitialization.
// 2. The value set in InternalStorageCluster.
// This order ensures backward compatibility.

Reply:
done